src/native/aws_signing.c

```diff
@@ -54,6 +54,9 @@ struct s_aws_sign_request_callback_data {
 static void s_cleanup_callback_data(struct s_aws_sign_request_callback_data *callback_data) {
 
     JNIEnv *env = aws_jni_get_thread_env(callback_data->jvm);
+    if (env == NULL) {
```

Comment:
Is that the same logic as Mike did for python? Like when JVM has gone, just silently exist? If so, can we add some comments? Like https://github.com/awslabs/aws-crt-python/blob/main/source/auth_credentials.c#L578

Reply:
A little reluctant to copy-and-paste the same comment 60 times.

Reply:
Can we add the reason behind at the PR description at least?

Reply:
I added the comments but still am not thrilled with the idea of every get-ptr-check-ptr-use-ptr requiring a comment on check ptr.